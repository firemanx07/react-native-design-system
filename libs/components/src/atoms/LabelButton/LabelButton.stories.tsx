import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import { StorybookScreen, StorybookSection } from '../../storybook';
import {
  LabelButton as LabelButtonBase,
  PropsType as LabelButtonProps,
  LabelButtonVariant,
} from './LabelButton';

const Story: ComponentMeta<typeof LabelButtonBase> = {
  component: LabelButtonBase,
  title: 'Atoms/Label Button',
  argTypes: {
    variant: {
      control: { type: 'radio' },
    },
  },
};
export default Story;

const Template: ComponentStory<typeof LabelButtonBase> = (
  args: LabelButtonProps
) => (
  <StorybookScreen>
    <StorybookSection>
      <LabelButtonBase {...args} />
    </StorybookSection>
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['variant', 'disabled', 'underline', 'children'],
  },
};

const defaultArgs: LabelButtonProps = {
  variant: LabelButtonVariant.Regular,
  disabled: false,
  underline: false,
  children: 'Label Button Text',
  onPress: () => {},
};

export const LabelButton = Template.bind({});
LabelButton.parameters = parameters;
LabelButton.args = {
  ...defaultArgs,
};
