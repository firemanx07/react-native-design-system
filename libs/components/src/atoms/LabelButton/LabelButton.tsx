import { styled } from '@proxym/themes';
import React, { memo } from 'react';
import { StyleProp, ViewStyle, TextProps } from 'react-native';

import { TestIDType } from '../../types';
import { BaseText, TextVariant, TextWeight } from '../BaseText';

export enum LabelButtonVariant {
  Regular = 'Regular',
  Medium = 'Medium',
  Small = 'Small',
}

export type PropsType = {
  children: TextProps['children'];
  onPress: () => void;
  variant: LabelButtonVariant;
  disabled?: boolean;
  underline?: boolean;
  style?: StyleProp<ViewStyle>;
} & TestIDType;

export const LabelButton = ({
  children,
  variant,
  underline,
  disabled,
  onPress,
  style,
  testID,
}: PropsType) => {
  return (
    <Container
      onPress={onPress}
      disabled={disabled}
      style={style}
      testID={testID}
    >
      <ButtonText
        buttonVariant={variant}
        disabled={disabled}
        underline={underline}
      >
        {children}
      </ButtonText>
    </Container>
  );
};

const labelButtonTextVariant = {
  [LabelButtonVariant.Regular]: TextVariant.H2,
  [LabelButtonVariant.Medium]: TextVariant.L,
  [LabelButtonVariant.Small]: TextVariant.M,
};

type ContainerType = {
  disabled?: boolean;
};

const Container = styled.TouchableOpacity<ContainerType>`
  align-self: flex-start;
  padding-horizontal: ${({ theme }) => theme.ds.spacing.small}px;
  padding-vertical: ${({ theme }) => theme.ds.spacing.tiny}px;
  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
`;

type ButtonTextType = {
  buttonVariant: LabelButtonVariant;
  underline?: boolean;
};

export const ButtonText = styled(BaseText).attrs(
  ({ buttonVariant }: ButtonTextType) => ({
    variant: labelButtonTextVariant[buttonVariant],
    weight: TextWeight.Bold,
  }),
)<ButtonTextType>`
  text-decoration-line: ${({ underline }) =>
    underline ? 'underline' : 'none'};
`;

export default memo(LabelButton);
