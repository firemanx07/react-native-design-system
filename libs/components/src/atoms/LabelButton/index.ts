export { default as LabelButton, LabelButtonVariant } from './LabelButton';
