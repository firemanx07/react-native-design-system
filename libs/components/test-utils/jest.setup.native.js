import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
