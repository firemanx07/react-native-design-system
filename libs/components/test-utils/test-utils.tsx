import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
