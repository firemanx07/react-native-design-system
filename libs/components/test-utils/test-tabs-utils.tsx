import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
