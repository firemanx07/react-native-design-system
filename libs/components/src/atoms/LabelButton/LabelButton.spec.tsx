import { testAllStories } from '../../../test-utils';
import * as stories from './LabelButton.stories';

describe('LabelButton tests', () => {
  testAllStories(stories);
});
