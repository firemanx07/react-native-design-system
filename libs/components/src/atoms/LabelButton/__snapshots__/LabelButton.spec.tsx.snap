// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelButton tests LabelButton snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "flex-start",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 4,
        }
      }
    >
      <Text
        buttonVariant="Regular"
        disabled={false}
        fontFamily="Rubik-Bold"
        fontSize={18}
        lineHeight={24}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 18,
              "lineHeight": 24,
            },
            {
              "textDecorationLine": "none",
            },
          ]
        }
        underline={false}
      >
        Label Button Text
      </Text>
    </View>
  </View>
</View>
`;
